Update IoT Hub sample to use new Events Hubs API (#15)
Azure-Samples/azure-iot-samples-java: iot-hub/Quickstarts/read-d2c-messages/README.md

the details required to form the Event Hubs compatible connection string.

- `az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}`
- `az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}`
- `az iot hub policy show --name service --query primaryKey --hub-name {your IoT Hub name}`

## Checkpointing

For an example that uses checkpointing, follow up this sample with the [sample that uses
Azure Storage Blob to create checkpoints](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java).

Note that this requires adding a new dependency in your `pom.xml` to use the [Azure Storage Blob checkpoint store](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md).

## WebSocket and proxy

If using WebSockets, configure the `EventHubClientBuilder` to use transport type `AmqpTransportType.AMQP_WEB_SOCKETS`.

> If your application runs behind a proxy server, then you need to configure the proxy options and set the transport 
> type to `AmqpTransportType.AMQP_WEB_SOCKETS` as shown in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java). 

Reviewer: I'd recommend re-structuring as below to put emphasis on proxy options

```suggestion
If your application runs behind a proxy server, then apart from setting the transport type to `AmqpTransportType.AMQP_WEB_SOCKETS`,
you need to configure the proxy options as shown in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java). 
```

Reviewer 2: Hey Ramya -- I think 'apart' is confusing. Do you mean this:

If your application runs behind a proxy server, then, in addition to setting the transport type to `AmqpTransportType.AMQP_WEB_SOCKETS`, you also need to configure the proxy options as shown in the `setupProxy` method in [the sample](./src/main/java/com/microsoft/docs/iothub/samples/ReadDeviceToCloudMessages.java).

Reviewer 2: @srnagar

Reviewer: yes, that sounds much better :)

- `az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}`
- `az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}`
- `az iot hub policy show --name service --query primaryKey --hub-name {your IoT Hub name}`

## Checkpointing

For an example that uses checkpointing, follow up this sample with the [sample that uses
Azure Storage Blob to create checkpoints](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java).

Note that this requires adding a new dependency in your `pom.xml` to use the [Azure Storage Blob checkpoint store](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md).

## WebSocket and proxy

If using WebSockets, configure the `EventHubClientBuilder` to use transport type `AmqpTransportType.AMQP_WEB_SOCKETS`.

If your application runs behind a proxy server, then, in addition to setting the transport type to 

> AmqpTransportType.AMQP_WEB_SOCKETS, you also need to configure the proxy options as shown in the setupProxy method in 

Reviewer 2: @srnagar  Should have tics ( ` ) around  AmqpTransportType.AMQP_WEB_SOCKETS and setupProxy to give code formatting.  

(PS: I'd do this myself, but I haven't really messed much with making changes in other people's forks and don't want to stomp all over your stuff)

Author: Done

Reviewer 2: Awesome! Thanks!
